### PR TITLE
skip parts that are missing version1 fields

### LINKED ISSUE #70: skip parts without version1 fields
Log error and skip part when:
- required version1 fields are missing

_Originally suggested by @yulric in https://github.com/Big-Life-Lab/PHES-ODM-Validation/pull/62#discussion_r1026739168_

### PR DESCRIPTION
**Pull request by zargot**
when generating a schema for v1, all parts need to have version1 fields

fixes #70.

### CODE REVIEW

**Review by yulric — CHANGES_REQUESTED** [review]
The changes look good but I think we should return a warnings field for the generate_validation_schema function like we do for validate_data. Right now the only warning would be for an invalid version 1 part. Do you want to spec that out in this PR or should I?

File: src/odm_validation/part_tables.py [hunk 1]
```diff
@@ -247,8 +251,12 @@ def filter_compatible(parts: Dataset, version: Version) -> Dataset:
     """Filters `parts` by `version`."""
     result = []
     for row in parts:
-        if not (is_compatible(row, version) or has_mapping(row, version)):
-            warning(f'skipping incompatible part: {get_partID(row)}')
+        pid = get_partID(row)
```
[the remark this answers is not in the document]
Comment (zargot) [reply src/odm_validation/part_tables.py]: done

**Comment by zargot** [comment]
waiting for spec for warning-list.

**Comment by zargot** [comment]
@yulric Ah, I forgot about this one. Yeah, I think we can merge it. Can you approve it?

**Review by yulric — APPROVED** [review]

**Pull request merged**